# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.cpp
# Commit: 5661dec1e128
# Full Hash: 5661dec1e128f48fc62be6ebc2774fc247b71fc3
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231588
# ==============================================================================

diff -r 15ccf86bce5a -r 5661dec1e128 layout/base/PresShell.cpp
--- a/layout/base/PresShell.cpp	Wed May 07 15:05:08 2025 +0000
+++ b/layout/base/PresShell.cpp	Wed May 07 15:05:09 2025 +0000
@@ -5063,7 +5063,8 @@
         // if the node is within the range, append it to the temporary list
         bool before, after;
         nsresult rv =
-            RangeUtils::CompareNodeToRange(content, aRange, &before, &after);
+            RangeUtils::CompareNodeToRange<TreeKind::ShadowIncludingDOM>(
+                content, aRange, &before, &after);
         if (NS_SUCCEEDED(rv) && !before && !after) {
           itemToInsert = i;
           bool snap;
